cmd/gateway/s3/gateway-s3.go

```diff
@@ -489,6 +498,29 @@ func (l *s3Objects) CopyObject(ctx context.Context, srcBucket string, srcObject
 	if _, err = l.Client.CopyObject(srcBucket, srcObject, dstBucket, dstObject, srcInfo.UserDefined); err != nil {
 		return objInfo, minio.ErrorRespToObjectError(err, srcBucket, srcObject)
 	}
+
+	var tagMap map[string]string
+	var tagStr string
+	var ok bool
+	tagStr, ok = srcInfo.UserDefined[xhttp.AmzObjectTagging]
+	if ok {
+		if tagMap, err = getTagMap(tagStr); err != nil {
+			return objInfo, minio.ErrorRespToObjectError(err, srcBucket, srcObject)
+		}
+	} else {
+		if tagStr, err = l.Client.GetObjectTagging(srcBucket, srcObject); err != nil {
+			return objInfo, minio.ErrorRespToObjectError(err, srcBucket, srcObject)
+		}
+		if tagMap, err = getTagMapParse(tagStr); err != nil {
+			return objInfo, minio.ErrorRespToObjectError(err, srcBucket, srcObject)
+		}
+	}
+	if len(tagMap) > 0 {
+		if err = l.Client.PutObjectTagging(dstBucket, dstObject, tagMap); err != nil {
+			return objInfo, minio.ErrorRespToObjectError(err, dstBucket, dstObject)
+		}
+	}
+
```

Comment:
Here also, you can add the tagging related headers in CopyObject API call itself, and let the target S3 server decide on how to handle the tags